calico/election.py

```diff
@@ -184,22 +220,26 @@ def _become_master(self):
                   self._key, self.id_string)
 
         self._master = True
-
-        while not self._stopped:
-            try:
-                self._etcd_client.write(self._key,
-                                        self.id_string,
-                                        ttl=self._ttl,
-                                        prevValue=self.id_string,
-                                        timeout=self._interval)
-            except Exception as e:
-                # This is a pretty broad except statement, but anything going
-                # wrong means this instance gives up being the master.
-                self._master = False
-                self._log_exception("renew master role", e)
-                raise RestartElection()
-
-            eventlet.sleep(self._interval)
+        try:
+            while not self._stopped:
+                try:
+                    _log.info("Refreshing master role")
+                    self._etcd_client.write(self._key,
+                                            self.id_string,
+                                            ttl=self._ttl,
+                                            prevValue=self.id_string,
+                                            timeout=self._interval/3)
+                    _log.info("Refreshed master role")
+                except Exception as e:
+                    # This is a pretty broad except statement, but anything
+                    # going wrong means this instance gives up being the
+                    # master.
+                    self._log_exception("renew master role", e)
+                    raise RestartElection()
+                eventlet.sleep(self._interval)
+        finally:
+            _log.info("Exiting master refresh loop, no longer the master")
+            self._master = False
```

Comment:
This is a bit weird. We raise the exception in the finally that is the only possible exception from the inner try.

Could we not just get rid of the outer `try`, and put self._master (and the log) just before the first `raise RestartElection`? Am I missing something?